src/components/cl/hier/cl_hier_coll.c

```diff
@@ -22,14 +23,16 @@ ucc_status_t ucc_cl_hier_coll_init(ucc_base_coll_args_t *coll_args,
     switch (coll_args->args.coll_type) {
     case UCC_COLL_TYPE_ALLREDUCE:
         return ucc_cl_hier_allreduce_rab_init(coll_args, team, task);
-    case UCC_COLL_TYPE_BARRIER:
-        return ucc_cl_hier_barrier_init(coll_args, team, task);
     case UCC_COLL_TYPE_ALLTOALL:
         return ucc_cl_hier_alltoall_init(coll_args, team, task);
     case UCC_COLL_TYPE_ALLTOALLV:
         return ucc_cl_hier_alltoallv_init(coll_args, team, task);
+    case UCC_COLL_TYPE_BARRIER:
+        return ucc_cl_hier_barrier_init(coll_args, team, task);
     case UCC_COLL_TYPE_BCAST:
         return ucc_cl_hier_bcast_2step_init(coll_args, team, task);
+    case UCC_COLL_TYPE_REDUCE:
+        return ucc_cl_hier_reduce_2step_init(coll_args, team, task);
```

Comment:
qq, why do we call specific alg init func from CL init func()?  is this default?

Reply:
yes, it's actually the only reduce implementation in CL HIER. However it calls sbgp reduces which can come from different TLs.